src/Zoo.cpp

```diff
@@ -0,0 +1,261 @@
+#include "Zoo.h"
+
+void Zoo::stopAnimal(int row, int col) const
+{
+    Location l(row, col);
+    for (const auto& animal : animals) {
+        if (animal->getLocation() == l)
+        {
+            animal->stop();
+        }
+    }
+}
+
+void Zoo::moveAnimal(int row, int col) const
+{
+    Location l(row, col);
+    for (const auto& animal : animals) {
+        if (animal->getLocation() == l) {
+            animal->move();
+        }
+    }
+}
+
+void Zoo::stepAllAnimals()
+{
+    for (const auto& animal : animals) {
+        animal->step();
+    }
+}
+
+Zoo::Zoo()
+{
+    std::srand(std::time(nullptr)); // seed the random number generator with the current time
+}
+
+void Zoo::run() {
+
+    initZoo();
+    std::string input = "start";
+    int row, col;
+    std::string animalType, animalName;
+
+    while (input != "exit")
+    {
+        printTable();
+        printAnimals();
+        printCommandNames();
+
+        std::cin >> input;
+
+        if (input == "stop") {
+            std::cout << "Enter a row and column number\n";
+            std::cin >> row;
+            std::cin >> col;
+            stopAnimal(row, col);
+        }
+        else if (input == "move") {
+            std::cout << "Enter a row and column number\n";
+            std::cin >> row;
+            std::cin >> col;
+            moveAnimal(row, col);
+        }
+        else if (input == "create") {
+            std::cout << "Enter the type of animal\n";
+            std::cin >> animalType;
+            std::cout << "Enter the name of animal\n";
+            std::cin >> animalName;
+            std::unique_ptr<Animal> newAnimal = createAnimal(animalType, animalName);
+            if (newAnimal != nullptr)
+            {
+                animals.emplace_back(std::move(newAnimal));
+            }
+            else
+            {
+                std::cout << "Unknown animal type: " << animalType << std::endl;
+            }
+        }
+        else if (input == "del") {
+            std::cout << "Enter a row and column number\n";
+            std::cin >> row;
+            std::cin >> col;
+            removeAnimal(row, col);
+        }
+        else if (input == "delAll") {
+            std::cout << "Enter the type of animal\n";
+            std::cin >> animalType;
+            deleteAnimalsOfType(animalType);
+        }
+        else if (input == "help") {
+            printHelpMenu();
+        }
+        else if (input == "exit") {
+            break;
+        }
+        else if (input == ".") {
+           
+        }
+        else {
+            std::cout << "Invalid command\n";
+        }
+        stepAllAnimals();
+    }
+}
+
```

Comment:
class zoo should not be used to manage the board, you must create a seperate file for managing the board